packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -87,4 +112,47 @@ export class DefaultHasOneRepository<
     }
     return found[0];
   }
+
+  async delete(where?: Where<TargetEntity>, options?: Options): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return targetRepository.deleteAll(
+      constrainWhere(where, this.constraint as Where<TargetEntity>),
+      options,
+    );
+  }
+
+  async patch(
+    dataObject: DataObject<TargetEntity>,
+    options?: Options,
+  ): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return await targetRepository.updateAll(
+      dataObject,
+      constrainWhere({}, this.constraint as Where<TargetEntity>),
+    );
+  }
+
+  async put(
+    dataObject: DataObject<TargetEntity>,
+    options?: Options,
+  ): Promise<void> {
+    const targetRepository = await this.getTargetRepository();
+    const entity = targetRepository.entityClass;
+    const primaryKeys = targetRepository.entityClass.getPrimaryKey(entity);
+    let id = (this as AnyObject)[''];
+    if (primaryKeys.length === 1 || primaryKeys.length === 0) {
+      id = Reflect.get(dataObject, primaryKeys[0]);
+    } else if (primaryKeys.length > 1) {
+      for (let idx = 0; idx < primaryKeys.length; idx++) {
```

Comment:
I didn't see a test case for multiple keys

Reply:
Same answer as above related to PUT (removed and will be handled in a new PR)